andela/lotus-ah
# #160451606 Implement user follow feature

#### What does this PR do?
    This PR create four routes for user follow feature. The routes functionalities are follow,unfollow, list all followers, list all following
#### Description of Task to be completed?
     1. Write test for all 4 routes
     2. Create the routes
     3. Create the controllers
#### How should this be manually tested?
- Start the app locally on port `3000`
- authenticate at least two users
- Construct follow url as follows ```/api/v1/profiles/<followingId>/follow```
- Make a `POST` request to the url
- You should get a response with the structure below:
```js
{
    "profile": {
        "username": null,
        "image": null,
        "bio": null,
        "following": false
    }
}
```
#### Any background context you want to provide?
      Allowing users to follow one another enhances the community mentality.
#### What are the relevant pivotal tracker stories?
      #160451606
#### Screenshots
<img width="1680" alt="screen shot 2018-10-16 at 7 12 45 pm" src="https://user-images.githubusercontent.com/43237059/47046424-f9b7d080-d18c-11e8-9ff8-634a67006f3a.png">

#### Questions:
   None
